code/datums/components/combustible_flooder.dm

```diff
@@ -1,26 +1,45 @@
+/// Component that floods gas when ignited by fire.
 /datum/component/combustible_flooder
 	// Gas type, molar count, and temperature. All self explanatory.
 	var/gas_id
 	var/gas_amount
 	var/temp_amount
 
-/datum/component/combustible_flooder/Initialize(gas_id, gas_amount, temp_amount)
+/datum/component/combustible_flooder/Initialize(initialize_gas_id, initialize_gas_amount, initialize_temp_amount)
 
-	src.gas_id = gas_id
-	src.gas_amount = gas_amount
-	src.temp_amount = temp_amount
+	src.gas_id = initialize_gas_id
+	src.gas_amount = initialize_gas_amount
+	src.temp_amount = initialize_temp_amount
 
 	RegisterSignal(parent, COMSIG_PARENT_ATTACKBY, .proc/attackby_react)
 	RegisterSignal(parent, COMSIG_ATOM_FIRE_ACT, .proc/flame_react)
+	RegisterSignal(parent, COMSIG_ATOM_BULLET_ACT, .proc/projectile_react)
+	RegisterSignal(parent, COMSIG_ATOM_TOOL_ACT(TOOL_WELDER), .proc/welder_react)
+	RegisterSignal(parent, COMSIG_ATOM_SECONDARY_TOOL_ACT(TOOL_WELDER), .proc/welder_react)
+	if(isturf(parent))
+		RegisterSignal(parent, COMSIG_TURF_EXPOSE, .proc/hotspots_react)
 
 /datum/component/combustible_flooder/UnregisterFromParent()
 	UnregisterSignal(parent, COMSIG_PARENT_ATTACKBY)
 	UnregisterSignal(parent, COMSIG_ATOM_FIRE_ACT)
+	UnregisterSignal(parent, COMSIG_ATOM_BULLET_ACT)
+	UnregisterSignal(parent, COMSIG_ATOM_TOOL_ACT(TOOL_WELDER))
+	UnregisterSignal(parent, COMSIG_ATOM_SECONDARY_TOOL_ACT(TOOL_WELDER))
+	if(isturf(parent))
+		UnregisterSignal(parent, COMSIG_TURF_EXPOSE)
 
-/// Do the flooding.
-/datum/component/combustible_flooder/proc/flood(mob/user, temp_amount)
+/// Do the flooding. Trigger temperature is the temperature we will flood at if we dont have a temp set at the start. Trigger referring to whatever triggered it.
+/datum/component/combustible_flooder/proc/flood(mob/user, trigger_temperature)
+	var/delete = TRUE
```

Comment:
This only appears to delete the parent, you should clarify the variable name better.